File: src/pages/exercises/string-cipher/_codes/ruby/response/src/cipher.rb
class Cipher
  def Cipher.complent(rotate)
    complent = {}
    input_chars = 'abcdefghijklmnopqrstuvwxyz'.split('')
    output_chars = input_chars.rotate(rotate)
    input_chars.each_with_index do |char, index|
      complent[char.upcase] = output_chars[index].upcase;
      complent[char.downcase] = output_chars[index].downcase;
    end
    return complent    
  end
  
  def Cipher.rot(message, number = 13)
    substitution = complent(number)
    return message
      .split('')
      .map { |char| substitution[char] || char }
      .join('')
  end
end

File: src/pages/exercises/string-cipher/_codes/js/response/cipher.js
Array.prototype.rotate = function(n) {
  return this.slice(n, this.length).concat(this.slice(0, n));
};

function complent(rotate) {
  let complent = {};
  let inputChars = 'abcdefghijklmnopqrstuvwxyz'.split('');
  let outputChars = inputChars.rotate(rotate);

  inputChars.forEach((char, index) => {
    complent[char.toLowerCase()] = outputChars[index].toLowerCase();
    complent[char.toUpperCase()] = outputChars[index].toUpperCase();
  });

  return complent;
}

function rot(message, number = 13) {
  let substitution = complent(number);

  return message
    .split('')
    .map(char => substitution[char] || char)
    .join('');
}

export { rot };
